keps/sig-auth/541-external-credential-providers/README.md

```diff
@@ -127,24 +154,37 @@ be readable and executable by the client process.
 `args` specifies extra arguments passed to the executable.
 
 `env` specifies environment variables to pass to the provider. The environment
-variables set in the client process are not passed.
+variables set in the client process are also passed to the provider.
```

Comment:
Updated to reflect current behavior.

Reply:
Worth discussing if we should "fix" this.

Reply:
IMO passing env vars through is more intuitive than not. +1 to keeping the current behavior

Reply:
+1 to keeping the current behavior. curious to learn the thinking behind the original design, though.

Reply:
So I agree that the current behavior is more intuitive and convenient, but I am not sure it is _correct_.  For example, by default, `sudo` does not pass in the current environment to the executed command.  `-E` must be explicitly set:

```
       -E, --preserve-env
                   Indicates to the security policy that the user wishes to preserve their existing environment variables.  The security policy may return an error if the user does not have permission to preserve the environment.
```

My previous (limited) experience with `exec` based flows to retrieve credentials has involved not passing of environment variables as a security measure (though I am not sure what attacks were being defended against).

Reply:
FWIW, we would need to preserve env variables to migrate GCP auth plugin off. I'm ok with going with the current behavior.

Reply:
> FWIW, we would need to preserve env variables to migrate GCP auth plugin off. I'm ok with going with the current behavior.

Is there some reason that the environment variables could not be included within the `exec` section?

---

Overall consensus seems to be preserve environment variables which is fine with me.  Not breaking stuff and helping with the `gcp` migration seem like good enough reasons.